build.cmd

```diff
@@ -28,27 +32,48 @@ if '!_FloatingDependencies!'=='0' (
   set _defaultBuildConfig=/p:Configuration=Windows_NT_Debug
   )
 )
+:: *** end WCF Content ***
+
+:ReadArguments
+:: Read in the args to determine whether to run the native build, managed build, or both (default)
+set "__args=%*"
+if /i [%1] == [native] (set __buildSpec=native&&set "__args=%__args:~6%"&&shift&&goto Tools)
+if /i [%1] == [managed] (set __buildSpec=managed&&set "__args=%__args:~7%"&&shift&&goto Tools)
 
+:Tools
+:: Setup VS
 if not defined VisualStudioVersion (
     if defined VS140COMNTOOLS (
         call "%VS140COMNTOOLS%\VsDevCmd.bat"
-        goto :EnvSet
+        goto :Build
     )
 
-    if defined VS120COMNTOOLS (
-        call "%VS120COMNTOOLS%\VsDevCmd.bat"
-        goto :EnvSet
-    )
-
-    echo Error: build.cmd requires Visual Studio 2013 or 2015. 
-    echo        Please see https://github.com/dotnet/wcf/blob/master/Documentation/developer-guide.md for build instructions.
+    echo Error: build.cmd requires Visual Studio 2015.
+    echo        Please see https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/developer-guide.md for build instructions.
     exit /b 1
 )
 
-:EnvSet
-
+:Build
+:: Restore the Tools directory
 call %~dp0init-tools.cmd
 
+:: Call the builds
+if "%__buildSpec%"=="managed"  goto :BuildManaged
+
+:BuildNative
```

Comment:
We chatted and agreed to stay as close to CoreFx as possible with optionally some comments explaining this.